pylint/plugins/hass_enforce_type_hints.py

```diff
@@ -2502,24 +2502,36 @@ class ClassTypeHintMatch:
             base_class="WaterHeaterEntity",
             matches=[
                 TypeHintMatch(
-                    function_name="precision",
-                    return_type="float",
+                    function_name="current_operation",
+                    return_type=["str", None],
                 ),
                 TypeHintMatch(
-                    function_name="temperature_unit",
-                    return_type="str",
+                    function_name="current_temperature",
+                    return_type=["float", None],
                 ),
                 TypeHintMatch(
-                    function_name="current_operation",
-                    return_type=["str", None],
+                    function_name="is_away_mode_on",
+                    return_type=["bool", None],
+                ),
+                TypeHintMatch(
+                    function_name="max_temp",
+                    return_type="float",
+                ),
+                TypeHintMatch(
+                    function_name="min_temp",
+                    return_type="float",
                 ),
                 TypeHintMatch(
                     function_name="operation_list",
                     return_type=["list[str]", None],
                 ),
                 TypeHintMatch(
-                    function_name="current_temperature",
-                    return_type=["float", None],
+                    function_name="precision",
+                    return_type="float",
+                ),
+                TypeHintMatch(
+                    function_name="supported_features",
+                    return_type=["WaterHeaterEntityFeature", "int"],
```

Comment:
This is the only adjustment. The other items in the file are just sorted alphabetically.